ui/src/app/settings/views/Dhcp/DhcpList/DhcpList.js

```diff
@@ -12,61 +17,225 @@ import {
   device as deviceSelectors,
   machine as machineSelectors
 } from "app/base/selectors";
+import Button from "app/base/components/Button";
+import Code from "app/base/components/Code";
+import Col from "app/base/components/Col";
+import VanillaLink from "app/base/components/Link";
 import Loader from "app/base/components/Loader";
+import MainTable from "app/base/components/MainTable";
+import Pagination from "app/base/components/Pagination";
+import Row from "app/base/components/Row";
+import SearchBox from "app/base/components/SearchBox";
 import selectors from "app/settings/selectors";
 
-// Temporarily create a item component, this will be replaced with a table.
-const DhcpListItem = ({ id, name, node, subnet }) => {
-  const subnetItem = useSelector(state =>
-    selectors.subnet.getById(state, subnet)
-  );
-  const controllerItem = useSelector(state =>
-    controllerSelectors.getBySystemId(state, node)
-  );
-  const deviceItem = useSelector(state =>
-    deviceSelectors.getBySystemId(state, node)
-  );
-  const machineItem = useSelector(state =>
-    machineSelectors.getBySystemId(state, node)
-  );
-  const nodeItem = controllerItem || deviceItem || machineItem;
-  const nodeType =
-    (controllerItem && "controller") ||
-    (deviceItem && "device") ||
-    (machineItem && "machine");
-  return (
-    <li>
-      {name} {subnet && `(subnet: ${subnetItem.name})`}{" "}
-      {nodeItem && `(${nodeType}: ${nodeItem.hostname})`}
-    </li>
+const generateURL = url => `${process.env.REACT_APP_MAAS_URL}/${url}`;
+
+const getNode = (controllers, devices, machines, nodeId) => {
+  const controllerItem = controllers.find(
+    controller => controller.system_id === nodeId
   );
+  const deviceItem = devices.find(device => device.system_id === nodeId);
+  const machineItem = machines.find(machine => machine.system_id === nodeId);
+  return {
+    category:
+      (controllerItem && "controller") ||
+      (deviceItem && "device") ||
+      (machineItem && "machine"),
+    node: controllerItem || deviceItem || machineItem
+  };
 };
 
```

Comment:
Just a vague general comment here, it feels like there's a fair bit of complexity in this component, is there any way we can decompose it further? If you don't see any obvious way to manage that, feel free to ignore this.

Reply:
Most of the complexity is for showing the "applies to" links. I think I can move that part into another component which should hugely simplify this one.